qiskit/qobj/models/pulse_simulator.py

```diff
@@ -0,0 +1,149 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2019, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+"""The simulator qobj models."""
+
+from marshmallow.validate import Range, Length, Regexp
+
+from qiskit.validation import bind_schema, BaseSchema, BaseModel
+from qiskit.validation.fields import (Integer, String, Number, List, Nested, Boolean,
+                                      Dict, InstructionParameter)
+from qiskit.validation.validate import PatternProperties
+
+
+class PulseHamiltonianSchema(BaseSchema):
+    """Schema for PulseHamiltonian."""
+    # pylint: disable=redefined-builtin
+
+    # Required properties.
+    h_str = List(String(), validate=Length(min=1), required=True)
+    dim_osc = List(Integer(validate=Range(min=1)), required=True)
+    dim_qub = List(Integer(validate=Range(min=2)), required=True)
+    vars = Dict(validate=PatternProperties({
+        Regexp('^([a-z0-9])+$'): InstructionParameter()
+    }), required=True)
+
+    # Optional properties.
+
+
+class PulseNoiseModelSchema(BaseSchema):
+    """Schema for PulseNoiseModel."""
+
+    # Required properties.
+    qubit = List(
+        Dict(validate=PatternProperties({
+            Regexp('^([A-Za-z])+$'): Number()
+        })),
+        required=True)
+    oscillator = List(
+        List(Number, validate=Length(equal=2)),
+        required=True
+    )
+
+    # Optional properties.
+
+
+class PulseOdeOptionSchema(BaseSchema):
```

Comment:
I do not think this should live in the `terra` repository. These are specific to the `Aer` simulator and could differ from simulator to simulator. The model should live in the Aer repository and be used to validate the submitted parameters.